server/dataone_package.py

```diff
@@ -24,57 +29,95 @@ def create_resource_map(resmap_pid, eml_pid, file_pids):
-        str(user.get('lastName', '')))
-
-    creator = '<creator>\n{0}\n</creator>\n'.format(individual_name)
-    contact = '<contact>\n{0}\n</contact>\n'.format(individual_name)
-    dataset_close = '</dataset>\n'
-    eml_close = '</eml:eml>'
-
-    # Append the above xml together to form the EML document
-    xml = top+namespace+dataset+title+creator+contact+dataset_close+eml_close
-
+    # The surName in the EML record is taken by the user's last name. If
+    # it doesn't exist, try to use the first name.
+    lastName = user.get('lastName', 'None')
+    firstName = user.get('firstName', 'None')
+    ns = ET.Element('eml:eml')
+
+    ns.set('xmlns:eml', "eml://ecoinformatics.org/eml-2.1.1")
+    ns.set('xsi:schemaLocation', "eml://ecoinformatics.org/eml-2.1.1 eml.xsd")
+    ns.set('xmlns:stmml', "http://www.xml-cml.org/schema/stmml-1.1")
+    ns.set('xmlns:xsi', "http://www.w3.org/2001/XMLSchema-instance")
+    ns.set('scope', "system")
+    ns.set('system', "knb")
+    ns.set('packageId', eml_pid)
+
+    dataset = ET.SubElement(ns, 'dataset')
+    ET.SubElement(dataset, 'title').text = str(tale.get('title', ''))
+
+    creator = ET.SubElement(dataset, 'creator')
+    individual_name = ET.SubElement(creator, 'individualName')
+    ET.SubElement(individual_name, 'givenName').text = firstName
+    ET.SubElement(individual_name, 'surName').text = lastName
+
+    abstract = ET.SubElement(dataset, 'abstract')
+    description = get_tale_description(tale)
+    ET.SubElement(abstract, 'para').text = description if\
+        bool(len(description)) else 'None'
+
+    contact = ET.SubElement(dataset, 'contact')
+    ind_name = ET.SubElement(contact, 'individualName')
+    ET.SubElement(ind_name, 'givenName').text = firstName
+    ET.SubElement(ind_name, 'surName').text = lastName
+
+    # Add a <otherEntity> block for each object
+    for item_id in item_ids:
+        item = Item().load(item_id, user=user)
+        description = item['description']
+
+        other_entity = ET.SubElement(dataset, 'otherEntity')
+        ET.SubElement(other_entity, 'entityName').text = item['name']
+        ET.SubElement(other_entity, 'entityDescription').text = description if\
+            bool(len(description)) else 'None'
+
+        physical = ET.SubElement(other_entity, 'physical')
+        ET.SubElement(physical, 'objectName').text = item['name']
+        ET.SubElement(physical, 'size').text = str(item['size'])
+
+        format = get_file_format(item_id, user)
+        data_format = ET.SubElement(physical, 'dataFormat')
+        externally_defined = ET.SubElement(data_format, 'externallyDefinedFormat')
+        ET.SubElement(externally_defined, 'formatName').text = format
+
+        ET.SubElement(other_entity, 'entityType').text = 'Other'
+
+    # call decode to get the string representation instead of as a byte str
+    xml = ET.tostring(ns, encoding='UTF-8', xml_declaration=True, method='xml').decode()
```

Comment:
My editor says `xml_declaration` isn't an arg to `tostring()`. Is this just me having the wrong version of Python or the package?

Reply:
You might have the wrong version or the editor is incorrect. The definition in `ElementTree.py` follows
```
def tostring(element,
             encoding=None,
             method=None, *,
             short_empty_elements=True,
             xml_declaration=None):
```

Reply:
👍